src/screens/map-screen/map-screen.component.tsx

```diff
@@ -42,12 +51,57 @@ const MapScreen = () => {
 
   /**
    * This functions animates the map view to the input region
-   * @param {Object<Region>} region
+   * @param region The region to animate to
    */
-  const onCampusToggle = region => {
+  const onCampusToggle = (region: Region) => {
     mapRef.current.animateToRegion(region);
   };
 
+  /*
+   * Handles the event when the user pressed on the Building Location Button
+   *
+   * Sets the current location to whereever the user currently is, and then
+   * perform point-polygon collision detection to find which building the
+   * user is in.
+   */
+  const onBuildingLocationPress = (): void => {
+    getCurrentLocationAsync().then(response => {
```

Comment:
to temporarily test what would have happened if you were in a building, simply do the following:
```
response.coords.latitude = [lat];
response.coords.longitude = [long];
```
where [lat] and [long] are coordinate within a given building